packages/ckeditor5-coremedia-blocklist/src/blocklistediting.ts

```diff
@@ -0,0 +1,319 @@
+
+    // Wait for the service to be available
+    this.#blocklistServiceSubscription = serviceAgent
+      .observeServices<BlocklistService>(createBlocklistServiceDescriptor())
+      .subscribe(onServiceRegisteredFunction);
+  }
+
+  /**
+   * Updates the internal list of blocked words and all affected markers whenever
+   * the blocklist from the serviceAgent blocklistService changes.
+   * @param blocklistService - the blocklist service
+   * @private
+   */
+  async #listenForBlocklistChanges(blocklistService: BlocklistService): Promise<void> {
+    blocklistService.observe_blocklist().subscribe({
+      next: (blockedWords: string[]) => {
+        // Get all words that differ from blockedWords and internalBlocklist
+        const { addedWords, removedWords } = this.#getAddedAndRemovedWords(blockedWords, this.internalBlocklist);
+
+        addedWords.forEach((word) => {
+          this.#addMarkersForWord(word);
+        });
+
+        removedWords.forEach((word) => {
+          this.#removeMarkersForWord(word);
+        });
+
+        // Set new value in internal list
+        if (addedWords.length > 0 || removedWords.length > 0) {
+          this.internalBlocklist = [...blockedWords];
+        }
+      },
+    });
+
+    const initialWords = await blocklistService.getList();
+    initialWords.forEach((word) => {
+      this.#addMarkersForWord(word);
+    });
+    this.internalBlocklist = [...initialWords];
+  }
+
+  /**
+   * Compares 2 string lists and returns the differences between the lists.
+   * All words, that exist in the newList, but not in the oldList, are handled as "added".
+   * All words in the oldList, that do not exist in the newList, are handled as "removed".
+   *
+   * When used to differentiate between the internally stored list and the list, provided
+   * by an external service, the oldList represents the internal list and the data from the
+   * service is used as the newList.
+   *
+   * @param newList - the more recent list
+   * @param oldList - the older list
+   * @returns an object, holding information about changes between the lists
+   * @private
+   */
+  #getAddedAndRemovedWords(newList: string[], oldList: string[]): { addedWords: string[]; removedWords: string[] } {
+    const addedWords = newList.filter((word) => !oldList.includes(word));
+    const removedWords = oldList.filter((word) => !newList.includes(word));
+    return { addedWords, removedWords };
+  }
+
+  /**
+   * Adds a word to the blocklist.
+   * Use this method for changes, triggered by the UI, not by the service.
+   *
+   * This method updates the internal list, changes the markers accordingly
+   * and triggers the blocklistService.
+   *
+   * @param wordToBlock - the word to add to the list
+   */
+  addBlocklistWord(wordToBlock: string): void {
+    // Update internal list
+    this.#addToInternalBlocklist(wordToBlock);
+    // Update markers
+    this.#addMarkersForWord(wordToBlock);
+    // Update value in service
+    serviceAgent
```

Comment:
As discussed in the pair review: This might cause an unexpected ui state if the service agent call returns with an error. After an error the words are still marked but not stored in the backend. 

Options: 
- Add markers after the words are synchronized (in the then block). 
- If the call returns with an error, remove the previously added markers.

Both ways have there UX advantages/disadvantages and are fine for me.